cpubars.c

```diff
@@ -173,15 +181,15 @@ static int cpustats_buf_size;
 void
 cpustats_init(void)
 {
-        if ((cpustats_fd = open("/proc/stat", O_RDONLY)) < 0)
-                epanic("failed to open /proc/stat");
-        if ((cpustats_load_fd = open("/proc/loadavg", O_RDONLY)) < 0)
-                epanic("failed to open /proc/loadavg");
+        if ((cpustats_fd = open(PROCPATH "/stat", O_RDONLY)) < 0)
+                epanic("failed to open " PROCPATH "/stat");
+        if ((cpustats_load_fd = open(PROCPATH "/loadavg", O_RDONLY)) < 0)
+                epanic("failed to open " PROCPATH "/loadavg");
 
         // Find the maximum number of CPU's we'll need
-        char *poss = read_all("/sys/devices/system/cpu/possible");
-        cpustats_cpus = cpuset_max(poss) + 1;
-        free(poss);
+        // This only gets the current number of CPUs online, precluding CPU
```

Comment:
I don't have much occasion to use cpubars these days, but when I wrote it part of the point was that it dealt nicely with CPU hotplug (top had a nasty habit of crashing when you hotplugged dozens of CPUs at once). So I'd like not to lose that.

If there's no way to get the maximum possible CPUs, I think it would be better to dynamically reallocate the `cpustats` structure in `cpustats_read` and deal with different-sized structures in `cpustats_subtract`.

Also, if this code goes away, you should also delete `cpuset_max` since it'll be unused.